### ISSUE TITLE
code is called twice

### ISSUE THREAD

**Issue by shunsuke0-byte** [emadeldeen24/ECGTransForm#6]
Hello, 
Could you explain why the self.evaluate(model, self.val_dl) is called twice in the code? 
```python
            self.evaluate(model, self.val_dl)
            tr_acc, tr_f1 = self.calc_results_per_run()
            # logging
            self.logger.debug(f'[Epoch : {epoch}/{self.hparams["num_epochs"]}]')
            for key, val in loss_avg_meters.items():
                self.logger.debug(f'{key}\t: {val.avg:2.4f}')
            self.logger.debug(f'TRAIN: Acc:{tr_acc:2.4f} \t F1:{tr_f1:2.4f}')

            # VALIDATION part
            self.evaluate(model, self.val_dl)
            ts_acc, ts_f1 = self.calc_results_per_run()
``` 

Additionally, on the terminal, I see the following output:

![image](https://github.com/emadeldeen24/ECGTransForm/assets/85477508/825d567b-52fb-4b00-850a-3f502d7bb29a)
Why are the accuracy (Acc) and F1 scores for both train and val exactly the same?

**Comment by shashank3009** [comment]
I think the first self.evaluate should be self.train_df (and not self.val_df) since we are already doing that below. And that is the reason why the train and val accuracy are coming the same. I have modified that part in my code.

**Comment by shunsuke0-byte** [comment]
thanks for reply :)

**Comment by huinee56** [comment]
@shunsuke0-byte Excuse me, would you be so kind as to share the dataset you have already downloaded with me? Currently, the dataset link provided by the author has expired. I would really appreciate your help.
